### PR TITLE
Transactions update

### PR DESCRIPTION
**Pull request by KaskMIL**
# In this new Pull Request I'm adding the following features:
- Add a new column to the animals' table.
- Insert new animals.
- Update the table using queries.
- Add queries to respond to the questions given
   - `BEGIN; UPDATE animals SET species = 'unspecified';`

Captura de Pantalla 2022-08-09 a la(s) 19 48 40

   - `ROLLBACK;`

Captura de Pantalla 2022-08-09 a la(s) 19 47 38

   - `UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';`

Captura de Pantalla 2022-08-09 a la(s) 19 51 57

   - `UPDATE animals SET species = 'Pokemon' WHERE name NOT LIKE '%mon';`

Captura de Pantalla 2022-08-09 a la(s) 19 55 49

   - `DELETE FROM animals;`

Captura de Pantalla 2022-08-09 a la(s) 20 03 25

   - `ROLLBACK;`

Captura de Pantalla 2022-08-09 a la(s) 20 04 55

   - `DELETE FROM animals WHERE date_of_birth > '2022-01-01';`

Captura de Pantalla 2022-08-09 a la(s) 20 12 28

   - `UPDATE animals SET weight_kg = weight_kg * -1;`

Captura de Pantalla 2022-08-09 a la(s) 20 12 41

   - `UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;`

Captura de Pantalla 2022-08-09 a la(s) 20 16 54

   - `SELECT COUNT(*) FROM animals;`

Captura de Pantalla 2022-08-09 a la(s) 20 22 19

   -  SOLUTION 1`SELECT neutered, MAX(escape_attempts) AS escapeAttempts FROM animals GROUP BY neutered;`

Captura de Pantalla 2022-08-09 a la(s) 22 57 14

   - SOLUTION 2 `SELECT name, escape_attempts, neutered FROM animals WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals) ORDER BY escape_attempts, neutered;`

Captura de Pantalla 2022-08-09 a la(s) 23 10 45

   - `SELECT AVG(weight_kg) FROM animals;`

Captura de Pantalla 2022-08-09 a la(s) 20 30 06

   - `SELECT neutered, COUNT(*) FROM animals GROUP BY neutered;`

Captura de Pantalla 2022-08-09 a la(s) 20 35 32

   - `SELECT species, MAX(weight_kg) AS maxWeight, MIN(weight_kg) AS minWeight FROM animals GROUP BY species;`

Captura de Pantalla 2022-08-09 a la(s) 20 43 47

   - `SELECT species, AVG(escape_attempts) AS avgscapes FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31' GROUP BY species;`

Captura de Pantalla 2022-08-09 a la(s) 20 48 41

   - `UPDATE animals SET date_of_birth = '2005-06-12' WHERE id = 8;`

Captura de Pantalla 2022-08-09 a la(s) 22 37 16

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,3 +4,8 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Gabumon', '2018-11-15', 2, TRUE, 8);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Pikachu', '2021-01-07', 1, FALSE, 15.04);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Devimon', '2017-05-12', 5, TRUE, 11);
+NSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', '2020-02-08', 0, FALSE, -11);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', '1993-04-02', 3, FALSE, -12.13);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', '2021-11-15', 2, TRUE, -5.7);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', '2005-06-07', 1, TRUE, -45);
```
Comment (tresorsawasawa) [inline data.sql]: - [ ] Kindly make sure you update `date_of_birth` for **Angemon** by using the correct date given in the **[project's requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Animal%3A%20His%20name%20is%20Angemon.%20He%20was%20born%20on%20Jun%2012th%2C%202005%2C%20and%20currently%20weighs%20%2D45kg.%20He%20is%20neutered%20and%20he%20has%20tried%20to%20escape%20once.)** as required for the project.

File: queries.sql [hunk 2]
```diff
@@ -7,4 +7,21 @@ SELECT date_of_birth FROM animals WHERE name = 'Augmon' OR name =  'Pikachu';
...
+DELETE FROM animals;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT first_delete;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO first_delete;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(*) FROM animals GROUP BY escape_attempts HAVING escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(*) FROM animals GROUP BY neutered;
```
Comment (tresorsawasawa) [inline queries.sql:26]: - [ ] Kindly make sure the query to answer this question: **"Who escapes the most, neutered or not neutered animals?"(`line 23`)** gives the correct answer(i.e it should give the animal with the most escape_attempts, neutered or not neutered).

**Review by tresorsawasawa — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @KaskMIL 
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KaskMIL** [comment]
Hi @tresorsawasawa 
Thanks for the review, I just update the requirements
Happy coding!

File: data.sql (hunk 1 above)
Comment (AnselemOdims) [inline data.sql]: - [ ] As pointed out by the previous reviewer, please ensure that the date for `Angemon` matches that in the requirement.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @KaskMIL , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: data.sql (hunk 1 above)
Comment (KaskMIL) [reply data.sql]: Hi @AnselemOdims 
I already update it, you can see the image at the bottom of the PR description

File: data.sql [hunk 3]
```diff
@@ -7,5 +7,5 @@ INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg)
 NSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Charmander', '2020-02-08', 0, FALSE, -11);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Squirtle', '1993-04-02', 3, FALSE, -12.13);
 INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Plantmon', '2021-11-15', 2, TRUE, -5.7);
-INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', '2005-06-07', 1, TRUE, -45);
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg) VALUES ('Angemon', '2005-06-12', 1, TRUE, -45);
```
Comment (mateo951) [inline data.sql:10]: Addressed ✔

**Review by mateo951 — APPROVED** [review]
### Hi @KaskMIL , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KaskMIL** [comment]
Hi @mateo951 
Thanks for your review!
Happy coding!

**Pull request merged**